packages/v3/src/esp-entity-table.ts

```diff
@@ -98,15 +99,16 @@ export class EntityTable extends LitElement implements RestAction {
           this.has_controls = true;
         }
         this.entities.push(entity);
-        this.entities.sort((a, b) =>
+        this.entities.sort((a, b) =>          
           a.entity_category < b.entity_category
             ? -1
             : a.entity_category == b.entity_category
-            ? a.name < b.name
+            ? a.sorting_weight < b.sorting_weight
```

Comment:
There is a matching native component? Wondering if both should be supported for now so that it uses the new sorting criteria when available and falls back to the original when not.
```
        this.entities.sort((a, b) => {
          const sortA = a.sorting_weight ?? a.name;
          const sortB = b.sorting_weight ?? b.name;
          return a.entity_category < b.entity_category
            ? -1
            : a.entity_category == b.entity_category
            ? a.name < b.name
            ? a.sorting_weight < b.sorting_weight
              ? -1
              : 1
            : 1
        });
```

Reply:
Added and seems to work
**With sorting weight and my sorting pr**
![with-weight-and-sorting-pr](https://github.com/esphome/esphome-webserver/assets/4093064/fac0cbba-86a4-4b28-853f-71de650671ea)
**Dev branch**
![without-weight-and-dev-branch](https://github.com/esphome/esphome-webserver/assets/4093064/429f1a98-c21b-4ec3-b6d7-65873e280510)